Update form-vs-fetcher.md (#7705)
remix-run/remix: docs/discussion/form-vs-fetcher.md

- **Updating a Single Field**: Maybe a user wants to change the name of an item in a list or update a specific property of a record. This action is minor and doesn't necessitate a new page or URL.

> - **Deleting a Record from a List**: In a list view, if a user deletes an item, they likely expect to remain on the list view, while warning the user of the destructive action.

Reviewer: ```suggestion
- **Deleting a Record from a List**: In a list view, if a user deletes an item, they likely expect to remain on the list view, with that item no longer in the list.
```
